To-do list/Controllers/TaskListViewController.swift

```diff
@@ -28,15 +27,23 @@ class TaskListViewController: UIViewController,
         super.viewDidAppear(animated)
         
         taskApiHelper.executeTaskList { [weak self] result in
-            guard let self = self else {
-                return
-            }
-            if let result = result {
+            guard let self = self else { return }
+            var message = String()
+            
+            switch result {
+            case .success(let result):
                 self.tasks = result
                 self.tasksTableView.reloadData()
+            case .failure(let error):
+                message = self.changeErrorMessage(with: error)
+            }
+            
+            if !message.isEmpty {
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
```

Comment:
1. message won't be empty only inside .failure block. 
2. Move `showAlert` code inside .failure block
3. Remove DispatchQueue.main logic. You API class callback should already execute on the main thread.